File: LondonFhirService.Core/Services/Foundations/FhirRecordDifferences/IFhirRecordDifferenceService.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using LondonFhirService.Core.Models.Foundations.FhirRecordDifferences;

namespace LondonFhirService.Core.Services.Foundations.FhirRecordDifferences
{
    public interface IFhirRecordDifferenceService
    {
        ValueTask<FhirRecordDifference> AddFhirRecordDifferenceAsync(
            FhirRecordDifference fhirRecordDifference,
            CancellationToken cancellationToken = default);
        ValueTask<IQueryable<FhirRecordDifference>> RetrieveAllFhirRecordDifferencesAsync(CancellationToken cancellationToken = default);
        ValueTask<FhirRecordDifference> RetrieveFhirRecordDifferenceByIdAsync(
            Guid fhirRecordDifferenceId,
            CancellationToken cancellationToken = default);
        ValueTask<FhirRecordDifference> ModifyFhirRecordDifferenceAsync(
            FhirRecordDifference fhirRecordDifference,
            CancellationToken cancellationToken = default);
        ValueTask<FhirRecordDifference> RemoveFhirRecordDifferenceByIdAsync(
            Guid fhirRecordDifferenceId,
            CancellationToken cancellationToken = default);
    }
}

File: LondonFhirService.Manage.Client/src/services/foundations/fhirRecordDifferences/iFhirRecordDifferenceService.ts
import type { FhirRecordDifference } from "../../../models/foundations/fhirRecordDifferences/FhirRecordDifference";
import type { FhirRecordDifferenceQuery } from "../../../models/foundations/fhirRecordDifferences/FhirRecordDifferenceQuery";

export interface IFhirRecordDifferenceService {
    retrieveFhirRecordDifferencesAsync(
        fhirRecordDifferenceQuery: FhirRecordDifferenceQuery,
        abortSignal?: AbortSignal): Promise<FhirRecordDifference[]>;

    retrieveFhirRecordDifferenceByIdAsync(
        fhirRecordDifferenceId: string,
        abortSignal?: AbortSignal): Promise<FhirRecordDifference>;

    modifyFhirRecordDifferenceAsync(
        fhirRecordDifference: FhirRecordDifference): Promise<FhirRecordDifference>;
}
